configure.ac

```diff
@@ -3229,6 +3229,7 @@ AC_CHECK_FUNCS(\
 	memmove \
 	memrchr \
 	memset \
+	memset_s \
```

Comment:
This availability detection (and/or its use cases) would not work "as is" AFAICT because [C11](https://en.cppreference.com/w/c/string/byte/memset) says `As with all bounds-checked functions, memset_s is only guaranteed to be available if __STDC_LIB_EXT1__ is defined by the implementation and if the user defines __STDC_WANT_LIB_EXT1__ to the integer constant 1 before including string.h.`

It feels like Squid should do the above if we want to give the standard memset_s() a chance to be properly detected/used.

Reply:
was able to detect with clang but gcc 9 did not. fair enough

Reply:
After the changes tried on a system undetected (Linux) and another when it was (MacOS).

Reply:
Does the following build command detect the functions existence on Linux?
```
 ./configure CXXFLAGS="-D__STDC_WANT_LIB_EXT1__=1"
```

... and this one use the custom compat version on MacOS?
```
 ./configure CXXFLAGS="-D__STDC_WANT_LIB_EXT1__=0"
```

If not the detection may need to use an AC_CHECK_DECLS line with string.h as a dependency include.

Reply:
So FreeBSD, MacOS detects it (rightfully) but not in my Linux box (not present indeed). even the previous mean to detect was ok too same results.